File: data_types_project/cpp/strong_typing.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int number = 10;
    float converted = static_cast<float>(number);

    cout << "Integer: " << number << endl;
    cout << "Converted to float: " << converted << endl;

    // Invalid operation example:
    // string text = "Age: ";
    // cout << text + number << endl;

    cout << "C++ requires correct types or explicit conversion." << endl;

    return 0;
}

File: data_types_project/rust/strong_typing.rs
fn main() {
    let number: i32 = 10;
    let converted: f32 = number as f32;

    println!("Integer: {}", number);
    println!("Converted to float: {}", converted);

    // Invalid operation example:
    // let result = "Age: " + number;

    println!("Rust requires explicit type conversion.");
}